spec/controllers/api/v1/registrations_controller_spec.rb

```diff
@@ -0,0 +1,232 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe API::V1::RegistrationsController, type: :controller do
+  describe 'Filters' do
+    it { is_expected.not_to use_before_action(:verify_authenticity_token) }
+    it { is_expected.to use_before_action(:verify_oauth_application) }
+  end
+
+  describe 'POST#create', devise_mapping: true do
+    context 'given a valid oauth application' do
+      context 'given a valid request' do
+        context 'given password_confirmation param' do
```

Comment:
Not sure about the description here as "given valid params" would just work e.g. sending all required params. While in the next example, "given invalid params" you could just use wrong params e.g. `email: 'dev', password: '123456', password_confirmation: 'invalid'. Minor comment but it allows keeping tests simpler to maintain as adding a new param would not necessarily require to change the test in the future.